feat: canonical plugin

vinckr opened the pull request:
This adds a canonical link option for blog posts ect.

I put https://www.ory.sh as canonical base url. 

<!--
Please link the GitHub issue this pull request resolves in the format of `#1234`. If you discussed this change
with a maintainer, please mention her/him using the `@` syntax (e.g. `@aeneasr`).

If this change neither resolves an existing issue nor has sign-off from one of the maintainers, there is a
chance substantial changes will be requested or that the changes will be rejected.

You can discuss changes with maintainers either in the Github Discusssions in this repository or
join the [ORY Chat](https://www.ory.sh/chat).
-->

## Proposed changes

<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added necessary documentation within the code base (if
      appropriate).

## Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

aeneasr commented on gatsby-config.js:
```diff
@@ -156,7 +156,11 @@ module.exports = {
+      resolve: `gatsby-plugin-react-helmet-canonical-urls`,
+    options: {
+      siteUrl: `https://www.ory.sh`,
+    },
```
@vinckr you have not tried this out, because this includes a syntax error which does not allow the website to start. Please, when you make changes, try them out. Make sure they work. Reviewing PRs is very time and focus intense and it is frustrating to review stuff that does not work because the person has not actually tried if it works. I would even consider it rude. [Other ecosystems](https://www.neowin.net/news/linux-bans-university-of-minnesota-for-sending-buggy-patches-in-the-name-of-research/) outright ban contributors and their associates when broken things are shipped.

vinckr commented:
Sorry about that. 
I tested it but then I split up the PRs into two, 
one with the plugin and the other one with the content. 

But I did not know it errors out when you dont use the optional field anywhere and it is not explicitly typed. 
I used the dummy data in this PR too to test it out, so that is why it worked for me.
I should have tested it as is.

I added the blogpost into this PR now, so it works with `npm run develop`.

aeneasr commented:
Thank you!
